pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -509,6 +509,210 @@ def pow(self, z):
         return [PauliRot(self.data[0] * z, self.hyperparameters["pauli_word"], wires=self.wires)]
 
 
+class PCPhase(Operation):
+    r"""
+    A projector-controlled phase gate.
+
+    This gate applies a complex phase :math:`\phi` to first :math:`dim`
+    basis vectors of the input state while applying a complex phase :math:`- \phi`
+    to the remaining basis vectors. This is equivalent to applying a constant
+    phase shift to the subspace spanned by those :math:`dim` vectors. Consider the 2 qubit
+    case where :math:`dim = 2`.
+
+    .. math:: \Pi_{\phi}(\phi, dim=2, wires=[0,1]) = \begin{bmatrix}
+                e^{i\phi} & 0 & 0 & 0 \\
```

Comment:
Basically "weird" examples are often better at explaining the breadth of possibilities. Using dim=2 could be confusing since this corresponds to a single-qubit subspace

Reply:
Changed to dim = 3 :+1: